presto-main/src/main/java/com/facebook/presto/operator/ExistsOperator.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.FixedWidthBlock;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.plan.PlanNodeId;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slices;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class ExistsOperator
```

Comment:
It would complicate things a bit. Take look at `AddExchanges#visitJoin`, there is a check if a right side of join is scalar subquery or exists subquery if so then a broadcast join is used. Having this modeled as you suggested would make this check to be complicated and vulnerable to bugs. 

But from the other side that model would enable broadcast join to be applied to broader space of queries. I will do that.